orchagent/vnetorch.cpp

```diff
@@ -481,23 +697,31 @@ void VNetRouteOrch::handleRoutes(const Request& request)
         {
             ifname = request.getAttrString(name);
         }
+        else if (name == "nexthop")
+        {
+            auto ipstr = request.getAttrString(name);
+            ip_addresses = IpAddresses(ipstr);
+        }
         else
         {
-            SWSS_LOG_WARN("Logic error: Unknown attribute: %s", name.c_str());
-            return;
+            SWSS_LOG_INFO("Unknown attribute: %s", name.c_str());
+            continue;
         }
     }
 
     const std::string& vnet_name = request.getKeyString(0);
     auto ip_pfx = request.getKeyIpPrefix(1);
+    auto op = request.getOperation();
+    nextHop nh = { ip_addresses, ifname };
 
-    SWSS_LOG_INFO("VNET-RT '%s' add for ip %s", vnet_name.c_str(), ip_pfx.to_string().c_str());
+    SWSS_LOG_INFO("VNET-RT '%s' op '%s' for ip %s", vnet_name.c_str(),
+                   op.c_str(), ip_pfx.to_string().c_str());
 
     if (vnet_orch_->isVnetExecVrf())
     {
-        if (!doRouteTask<VNetVrfObject>(vnet_name, ip_pfx, ifname))
+        if (!doRouteTask<VNetVrfObject>(vnet_name, ip_pfx, nh, op))
         {
-            throw std::runtime_error("Route add failed");
+            throw std::runtime_error("Route update failed");
```

Comment:
>  throw std::runtime_error("Route update failed"); [](start = 10, length = 50)

it looks to me it is dangerous to crash here. what if the nexthop interface is not created yet in the doRouteTask.

Reply:
I checked the routeorch.cpp, it does not throw error, if add or delete route fails.

---
In reply to: [253250419](https://github.com/Azure/sonic-swss/pull/766#discussion_r253250419) [](ancestors = 253250419)

Reply:
it just retry.

---
In reply to: [253250430](https://github.com/Azure/sonic-swss/pull/766#discussion_r253250430) [](ancestors = 253250430,253250419)